packages/dev/core/src/Engines/engine.ts

```diff
@@ -421,7 +421,19 @@ export class Engine extends ThinEngine {
     /**
      * Gets a boolean indicating if the pointer is currently locked
      */
-    public isPointerLock = false;
+    public get isPointerLock() {
+        // Return either manual pointerlock state or canvas pointerlock state
+        return this._isPointerLock || (this._isCanvasPresent && document.pointerLockElement === this._renderingCanvas);

```

Comment:
Unfortunately, this won't work in all cases. If someone puts `document = {}` somewhere in Babylon Native, which happens sometimes to work around issues, this logic will still cause `isPointerLock` to return true when it shouldn't. Maybe we need to check if pointerLockElement is valid also?

```suggestion
        return this._isPointerLock || (this._isCanvasPresent && document.pointerLockElement && document.pointerLockElement === this._renderingCanvas);
```